tests/logging/log_writer_test.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/io/log_writer.h>
+
+#include <aws/common/string.h>
+#include <aws/testing/aws_test_harness.h>
+
+#include <errno.h>
+#include <stdio.h>
+
+#ifndef WIN32
+#   include <sys/file.h>
+#endif // WIN32
+
+#define TEST_WRITER_MAX_BUFFER_SIZE 4096
+
+/*
+ * We need to add some filesystem support in common.  Hard-code for now.
+ */
+static const char *s_test_file_name =
+#ifdef WIN32
```

Comment:
mkstemp() or _mktemp_s()? Should be fine for a test.

Reply:
will switch

Reply:
Putting off in favor of long-term platform-specific file support